tidy3d/web/api/mode.py

```diff
@@ -137,6 +145,106 @@ def run(
         to_file=results_file, verbose=verbose, progress_callback=progress_callback_download
     )
 
+def run_batch(
+    mode_solvers: List[ModeSolver],
+    task_name: str = "BatchModeSolver",
+    folder_name: str = "BatchModeSolvers",
+    results_files: List[str] = None,
+    verbose: bool = True,
+    max_workers: int = DEFAULT_NUM_WORKERS,
+    max_retries: int = DEFAULT_MAX_RETRIES,
+    retry_delay: float = DEFAULT_RETRY_DELAY,
+    progress_callback_upload: Callable[[float], None] = None,
+    progress_callback_download: Callable[[float], None] = None
+) -> List[ModeSolverData]:
+    """
+    Submits a batch of ModeSolver to the server concurrently, manages progress, and retrieves results.
+
+    Parameters
+    ----------
+    mode_solvers : List[ModeSolver]
+        List of mode solvers to be submitted to the server.
+    task_name : str
+        Base name for tasks. Each task in the batch will have a unique index appended to this base name.
+    folder_name : str
+        Name of the folder where tasks are stored on the server's web UI.
+    results_files : List[str], optional
+        List of file paths where the results for each ModeSolver should be downloaded. If None, a default path based on the folder name and index is used.
+    verbose : bool
+        If True, displays a progress bar. If False, runs silently.
+    max_workers : int
+        Maximum number of concurrent workers to use for processing the batch of simulations.
+    max_retries : int
+        Maximum number of retries for each simulation in case of failure before giving up.
+    retry_delay : int
+        Delay in seconds between retries when a simulation fails.
+    progress_callback_upload : Callable[[float], None], optional
+        Optional callback function called when uploading file with ``bytes_in_chunk`` as argument.
+    progress_callback_download : Callable[[float], None], optional
+        Optional callback function called when downloading file with ``bytes_in_chunk`` as argument.
+
+
+    Returns
+    -------
+    List[ModeSolverData]
+        A list of ModeSolverData objects containing the results from each simulation in the batch. ``None`` is placed in the list for simulations that fail after all retries.
+    """
+    console = get_logging_console()
+
+    # Check type to make sure the user is submitting a list of mode solvers
+    if not all(isinstance(x, ModeSolver) for x in mode_solvers):
+        log.error("[red]Stopped running a batch of mode solver simulations. Please provide a list of mode solvers of type [cyan]`td.plugins.mode.mode_solver.ModeSolver` [red]to run a batch of mode solver simulations.")
```

Comment:
we usually just `raise ValueError`, let's change to that.

The error message could be more descriptive, such as:

`mode_solvers` passed to `mode.web.run_batch()` should be a list containing all `ModeSolver` instances. Can't run the batched mode solver.